basic root motion controller & snow shader enhancements BUNDLE

human3562 opened the pull request:
воркинг он зе плеер мувмент брос

human3562 committed:
implement basic 8-dir root motion controller

camera controls + player movement controls (wasd + shift)

human3562 commented on scenes/player_animations_test/dummy_character.gd:
Здесь происходит логика контроллера, коменты убиру позже и приведу в человеческий вид. но потом :)

human3562 commented:
@SanchoviesApm400 Прошу первичную приемку контроллера!! Тестовая сцена - scenes/player_animations_test/player_test_scene.tscn

![изображение](https://github.com/user-attachments/assets/88138ec3-b124-4ef7-9e0a-130df8bdc974)

human3562 committed:
add sprinting animations

shift now toggles full sprint mode
Alt is now for walk / jog

human3562 commented:
Обновил контроллер, добавил три анимации для полного бега

Shift - фулл спринт
Alt - ходить / микробег

@SanchoviesApm400

human3562 committed:
tweaks

changed up terra_test scene, some minor tweaks to player animations

human3562 commented on addons/terrain_3d/bin/~libterrain.windows.debug.x86_64.dll:
че за срань

SanchoviesApm400 reviewed (dismissed):
так-с ну неплохо неплохо

SanchoviesApm400 commented:
- [ ] добавить джампы ну просто мастхев ю ноу блин
- [ ] желательно еще возможность изменять приближение камеры 
- [ ] ну и пофиксить начальную анимацию бега назад (спринт назад) 
- [ ]  ускорить ходьбу и бег в анимациях бега и ходьбы вправо и влево соотвественно

human3562 committed:
speed up side animations

looks hella weird

human3562 committed:
Delete ~libterrain.windows.debug.x86_64.dll

human3562 committed:
Merge branch 'main' into player-controller-with-animations

human3562 reviewed (commented):
Добавляю доп-пункты:
- [ ] Заменить анимации ходьбы, легкого бега вправо и влево на адекватные
- [x] Персонаж падает со скоростью ну примерно половины перышка

human3562 commented:
В этот ПР попадут небольшие изменения с работой снега, ограничусь просто небольшими оптимизациями перед тем как продолжить в другой ветке

human3562 committed:
jumpi jumpi

add draft jumping logic, some tweaks to terrain deformation
camera changing logic
changed turn-rate on sprint

human3562 commented:
@SanchoviesApm400 нужно зачекать

human3562 commented:
- [ ] когда идет спам джампа исправить отсутствие анимации начала джампа
- [ ] При приземлении скорость сбрасывается в 0, нужна какая-то инерция

human3562 committed:
fall-velocity

human3562 committed:
update fall_with_velocity

human3562 committed:
Update .gitignore

human3562 committed:
remove snow blurring on movement
